src/script/interpreter.cpp

```diff
@@ -1680,6 +1712,100 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_INSPECTINPUT:
+                {
+                    // OP_INSPECTINPUT is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+
+                    if (stack.size() < 2)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    int n = CScriptNum(stacktop(-1), fRequireMinimal).getint();
+                    int idx = CScriptNum(stacktop(-2), fRequireMinimal).getint();
+                    popstack(stack);
+                    popstack(stack);
+
+                    auto inps = checker.GetTxvIn();
+                    auto spent_outputs = checker.GetPrecomputedTransactionData()->m_spent_outputs;
+                    if (idx < 0 || (unsigned int)idx >= inps->size() || (unsigned int)idx >= spent_outputs.size())
+                        return set_error(serror, SCRIPT_ERR_INTROSPECT_INDEX_OUT_OF_BOUNDS);
+                    const CTxIn inp = inps->at(idx);
+                    const CTxOut spent_utxo = spent_outputs[idx];
+
+                    switch (n)
+                    {
+                        case 0:
+                        {
+                            valtype vchOutpointFlag(1);
+                            vchOutpointFlag[0] = (unsigned char) ((!inp.assetIssuance.IsNull() << 7) + (inp.m_is_pegin << 6));
+                            stack.push_back(vchOutpointFlag);
+                            break;
+                        }
+                        case 1:
+                        {
+                            valtype vchPrevTxid;
+                            vchPrevTxid.insert(vchPrevTxid.begin(), inp.prevout.hash.begin(), inp.prevout.hash.begin() + 32);
+                            valtype vchPrevVout;
+                            auto vout_le = htole32(inp.prevout.n);
```

Comment:
for the avoidance of doubt can you do `uint32_t` rather than `auto` here?

I'm also a little uneasy about using `htole32` directly here but I see it's used already in `OP_DETERMINISTICRANDOM` so ok.

Reply:
Ah I see `htole32` is encapsulated in the next commit into its own `push4_le` method. Ok, I'm happy with that.